File: 3555-final-array-state-after-k-multiplication-operations-i/final-array-state-after-k-multiplication-operations-i.cpp
class Solution {
public:
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) {
        priority_queue<vector<int>, vector<vector<int>>, greater<vector<int>>> pq;
        for (int i = 0; i < nums.size(); i++)
            pq.push({nums[i], i});
        while (k > 0){
            vector<int> temp = pq.top();
            pq.pop();
            nums[temp[1]] *= multiplier;
            pq.push({nums[temp[1]], temp[1]});
            k -= 1;
        }
        return nums;
    }
};

File: 3555-final-array-state-after-k-multiplication-operations-i/final-array-state-after-k-multiplication-operations-i.java
class Solution {
    public int[] getFinalState(int[] nums, int k, int multiplier) {
        PriorityQueue<Pair<Integer, Integer>> pq = new PriorityQueue<Pair<Integer, Integer>>((p,q) -> {
            int valueCompare = Integer.compare(p.getKey(), q.getKey());
            if (valueCompare == 0)
                return Integer.compare(p.getValue(), q.getValue());
            return valueCompare;
            });
        for (int i = 0; i < nums.length; i++)
            pq.add(new Pair<Integer, Integer>(nums[i], i));

        while (k > 0){
            Pair<Integer, Integer> temp = pq.poll();
            nums[temp.getValue()] *= multiplier;
            pq.add(new Pair<Integer, Integer>(nums[temp.getValue()], temp.getValue()));
            k -= 1;
        }
        return nums;
    }
}